File: P2/P2_Player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Player
  @@MAXLEVEL = 10
  def initialize(nombre, nivel, enemigo, pBC, muerte = true, can_i_s=true)
    @name = nombre
    @level = nivel
    @dead = muerte
    @canISteal = can_i_s
    @enemy = enemigo
    @pendingBadConsequence = pBC
    @visibleTreasures = Array.new
    @hiddenTreasures = Array.new
  end
  
  def player(name)
    
    
  end
  
  def get_name
    
  end
  
  private
  def bring_to_life
    
  end
  
  private
  def getCombatLevel
   # each contador con if
  end
  
  private
  def incrementLevels(l)
    
  end
  
  private
  def decrementLevels(l)
    #que no se pasa de un si se pasa pues se pone a uno.
  end
  
  private 
  def setPendingBadConsequence(b)
    
  end
  
  private
  def applyPrize(m)
    
  end
  
  private
  def applyBadConsequence(m)
    
  end
  
  private 
  def canMakeTreasureVisible(t)
    
  end
  
  private 
  def howManyVisibleTreasures(tKind)
    #ech contando cuntos hay de un determinado tipo
  end
  
  private
  def dieIfNoTreasures
    #=true
  end
  
  def isDead
    @dead
  end
  
  def getHiddenTreasures
    
  end
  
  def getVisibleTreasures
    
  end
  
  def combat(m)
    
  end
  
  def makeTreasureVisible(t)
    
  end
  
  def discardVisibleTreasure(t)
    
  end
  
  def discardHiddenTreasure(t)
    
  end
  
  def validState
    
  end
  
  def initTreasures
    
  end
  
  def getLevels
    
  end
  
  def stealTreasure
    
  end
  
  def stealTreasure
    
  end
  
  def setEnemy(enemy)
    
  end
  
  def giveMeATreasure
    
  end
  
  def canISteal
    
  end
  
  private
  def canYouGiveMeATreasure
    #ver si array de tesoros si te puedo dar tesros si no está vacio
  end
  
  private
  def  haveStolen
    
  end
  
 def  discardAllTreasures
   
 end
  
end

File: P2/P2_Player.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author Dunia
 */
import java.util.ArrayList;


public class Player {
    final int MAXLEVEL = 10;
    private String name;
    private int level;
    private boolean dead = true;
    private boolean canISteal= true;
    private Player enemy;
    private BadConsequence pendingBadConsequence;
    private ArrayList<Treasure> visibleTreasures;
    private ArrayList<Treasure> hiddenTreasures;
    
    
    public Player(String name){
        
    }
    
    public String getName(){
        return name;
    }
   
    private void bringToLife(){
        dead = false;
        level = 1;
    }
    
    private int getCombatLevel(){
        //coger los tesoros visibles cuando llegues a 10 has ganado.
        //usar el for
        int bonus = 0;
        for(Treasure vt: visibleTreasures)
           bonus =  vt.getBonus() + bonus;
     
              
        return level+bonus;
    }
    
    private void incrementLevels(int l){
        //que no suba de 10 niveles
        level = level + l;
        if(level > 10)
            level = 10;
    }
    
    private void decrementLevels(int l){
        //no podemos bajar de uno con maxlevels
        level = level - l;
        if(level < 1)
            level = 1;
    }
    
    private void setPendingBadConsequence(BadConsequence b){
        //modficador basico badconsequence
    }
    
    private void applyPrize(Monster m) {
        
    }
    
    private void applyBadConsequence(Monster m){
        
    }
    
    private boolean canMakeTreasureVisible(Treasure t){
        
    }
    
    private int howManyVisibleTreasures(TreasureKind tKind){
        //for con un contador
        int cuentaVisibles = 0;
        for(Treasure vt: visibleTreasures)
            cuentaVisibles++;
        return cuentaVisibles;
        
    }
    
    private void dieIfNoTreasures(){
        //dead = true
        
    }
    
    
    public boolean isDead(){
        return dead;
        
    }
    
    public ArrayList<Treasure> getHiddenTreasures() {
        
    }
    
    
    public ArrayList<Treasure> getVisibleTreasures(){
        
    }
    
    public CombatResult combat(Monster m){
        
    }
    
    public void makeTreasureVisible(Treasure t){
        
    }
    
    public void discardVisibleTreasure(Treasure t){
        
    }
    
    public void discardHiddenTreasure(Treasure t){
        
    }
    
    public boolean validState(){
        //si el jugador puede pasar de turno cuando se cumplen esas 2 condiciones
        //su penbadConse está vacio con isEmpty de la clase bad consequence
        //y si el tamaño de hiddentreasures no es mayor que 4.
       //si devuelve false hay que decirle q no puede pasar
        
    }
    
    public void initTreasures(){
        
    }
    
    public int getLevels(){
        
    }
    
    public Treasure stealTreasure(){
        
    }
    
    public void setEnemy(Player enemy){
        
    }
    
    private Treasure giveMeATreasure(){
        
    }
    
    public boolean canISteal(){
        //consultor básico hay que decir si ùede o no robar
    }
    
    private boolean canYouGiveMeATreasure(){
        //comprueba si el array de tesoros esta vacio o no (negacion de isempty)
    }
    
    private void haveStolen(){
        //cambia el valor de canISteal lo pone a false
        canISteal = false;
    }
    
    public void discardAllTreasures(){
        
    }
    
    
}
